internal/controller/scanjob_controller.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	sbombasticv1alpha1 "github.com/rancher/sbombastic/api/v1alpha1"
+	"github.com/rancher/sbombastic/internal/handlers"
+	"github.com/rancher/sbombastic/internal/messaging"
+)
+
+// ScanJobReconciler reconciles a ScanJob object
+type ScanJobReconciler struct {
+	client.Client
+	Scheme    *runtime.Scheme
+	Publisher messaging.Publisher
+}
+
+// +kubebuilder:rbac:groups=sbombastic.rancher.io,resources=scanjobs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=sbombastic.rancher.io,resources=scanjobs/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=sbombastic.rancher.io,resources=scanjobs/finalizers,verbs=update
+
+// Reconcile reconciles a ScanJob object.
+func (r *ScanJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := logf.FromContext(ctx)
+	log.Info("Reconciling ScanJob")
+
+	scanJob := &sbombasticv1alpha1.ScanJob{}
+	if err := r.Get(ctx, req.NamespacedName, scanJob); err != nil {
+		if errors.IsNotFound(err) {
+			// Object not found, likely already deleted
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("unable to get ScanJob: %w", err)
+	}
+
+	// Check if resource is being deleted
+	if !scanJob.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	}
+
+	if scanJob.IsFailed() || scanJob.IsComplete() || scanJob.IsInProgress() {
+		return ctrl.Result{}, nil
+	}
```

Comment:
Right now this overlaps with `scanJob.IsInProgress()`, can we use this method? I think we're trying to achieve the same check, meaning that if a scanjob is already in progress the reconciler must exit

Reply:
you're right, I didn't read the code properly